Assets/_BForBoss/_VFX/Scripts/PeriodicallyRestartEffect.cs

```diff
@@ -0,0 +1,30 @@
+//-----------------------------------------------------------------------
+// <copyright file="PeriodicallyRestartEffect.cs" company="PERIGON GAMES">
+//     Copyright (c) PERIGON GAMES. 2020 All Rights Reserved
+// </copyright>
+//-----------------------------------------------------------------------
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.VFX;
+
+namespace BForBoss
+{
+    [ExecuteInEditMode]     // Remove this if you don't want the script to update in editor.
+    public class PeriodicallyRestartEffect : MonoBehaviour
+    {
+
+        [SerializeField] private float _wait_time = 1.0f;     // Time between shots.
+        [SerializeField] private float _curr_time = 0f;       // Current time in the timer.
+        
+        void Update()
+        {
+            _curr_time += Time.deltaTime;    // Increment timer.
+            if (_curr_time >= _wait_time)     // If wait_time is reached:
+            {
+                _curr_time = 0;                          // Reset timer.
+                GetComponent<VisualEffect>().Reinit();  // Reinitialize the VFX (do the effect).
```

Comment:
So whenever you call `GetComponent`, it combs through the "GameObjects" components at N time and tries to find it. So it can be a bit of a computational waste. 

So it's normally better to store it locally within the class on `Awake()`

Reply:
and i'm kinda curious, why are you reinit as opposed to `Play`?
https://docs.unity3d.com/2018.3/Documentation/ScriptReference/Experimental.VFX.VisualEffect.html

Reply:
That makes sense! I'll switch it over to getting a reference instead.

Reply:
Reinit() is used instead of Play() because using Play() wasn't resetting the VFX for me, it just continues to play it. The VFX is a single burst (no loops), so continuing to play it would just play the empty space after the initial burst.

There are quite a few reasons to consider how we want to do VFX in the long term (mostly art style and cost), so this made sense to me for right now.

Reply:
I take it back! Just tested out Play() and it works just fine.